src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,448 @@
+
+from PIL import Image
+from src.sparsezoo.utils.numpy import load_numpy_list
+
+
+class File:
+    """
+    Object to wrap around common files. Currently, supporting:
+    - numpy files
+    - onnx files
+    - yaml files
+    - markdown files
+    - json files
+    - csv files
+    - image files
+    """
+
+    def __init__(
+        self, name: str, path: Optional[str] = None, url: Optional[str] = None
+    ):
+        """
+        Constructor for a File object.
+
+        :param name: name of the File
+        :param path: path of the File
+        :param url: url of the File
+        """
+        self.name = name
+        self.url = url
+        self.path = path
+
+        # self.path can have any extension, including no extension.
+        # However, the File object also contains information
+        # About its loadable extensions.
+        # Loadable files can be read into the memory.
+        self.loadable_extensions = {
+            ".npz": self._validate_numpy,
+            ".onnx": self._validate_onnx,
+            ".yaml": self._validate_yaml,
+            ".md": self._validate_markdown,
+            ".json": self._validate_json,
+            ".csv": self._validate_csv,
+            ".jpg": self._validate_img,
+            ".png": self._validate_img,
+            ".jpeg": self._validate_img,
+        }
+
+    # TODO: Add support for various integrations
+    def validate(
+        self, strict_mode: bool = True, integration: Optional[str] = None
+    ) -> bool:
+        """
+        Validate whether the File object is loadable or not.
+
+        :param strict_mode: specifies the behavior of private `_validate_{}` methods:
+            - if strict_mode: method will raise ValueError on error
+            - if not strict_mode: method will raise warning on
+                error
+        :param integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :return: boolean flag; True if File instance is loadable, otherwise False
+        """
+        _, extension = os.path.splitext(self.path)
+
+        if extension in self.loadable_extensions.keys():
+            validation_function = self.loadable_extensions[extension]
+            validation_function(strict_mode=strict_mode)
+            return True
+
+        else:
+            return False
+
+    def _validate_numpy(self, strict_mode):
+        if not load_numpy_list(self.path):
+            self._throw_error(
+                error_msg="Numpy file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_onnx(self, strict_mode):
+        if not onnx.load(self.path):
+            self._throw_error(
+                error_msg="Onnx file could not been loaded properly",
+                strict_mode=strict_mode,
+            )
+
+    def _validate_yaml(self, strict_mode):
+        try:
+            with open(self.path) as file:
+                yaml.load(file, Loader=yaml.FullLoader)
```

Comment:
this won't work here because recipes with modifiers won't be registered in yaml outside of sparseml

Reply:
Let's discuss it during the sync.